fault-proof/tests/sync.rs

```diff
@@ -801,4 +804,215 @@ mod sync {
 
         Ok(())
     }
+
+    /// Tests the `should_attempt_to_resolve` flag logic for IN_PROGRESS games.
+    ///
+    /// Verifies that the flag is correctly set based on:
+    /// - Game status (IN_PROGRESS vs resolved)
+    /// - Deadline status (passed vs not passed)
+    /// - Parent resolution status (resolved vs IN_PROGRESS)
+    /// - Ownership (own game vs not own game)
```

Comment:
This doesn’t actually cover (reject) the case where a game isn’t owned, right? That’s probably more of a permissionless-mode scenario, so it’s not a priority to add if it’s not included.